### PR TITLE
Update for Bevy 0.7

### PR DESCRIPTION
**Pull request by rparrett**
Throwing this draft PR out here in case it's useful, or somebody wants to collaborate on this.

With this PR, the project compiles, but:

- ~~**Does not display tilemaps**~~
- ~~I had to remove `MapQuery::get_layer` due to a mysterious lifetime-related error~~
- ~~There are some warnings left to clean up.~~

### CODE REVIEW

**Comment by rparrett** [comment]
Okay, probably fixed the "not displaying anything at all" issue, but I have only tested the `map` example so far.

**Comment by StarArawn** [comment]
Thank you for working on this! 👍

**Comment by rparrett** [comment]
| example | result |
|-|-|
| accessing_tiles  | ok |
| animation        | ok |
| atlas_spacing    | ok |
| bench            | ok |
| color            | ok |
| dynamic_map      | ok |
| game_of_life     | :x: panic |
| hex_column       | ok |
| hex_row          | ok |
| improved_iso     | ok |
| iso_diamond      | ok |
| iso_staggered    | ok |
| layers           | ok |
| ldtk             | :x: blank screen, warning |
| map              | ok |
| random_map       | ok |
| remove_tiles     | ok |
| replace_material | :x: compiler error |
| sparse_tiles     | ok |
| texture_atlas    | ok |
| tiled            | ok |
| visibility       | ok |

**Comment by rparrett** [comment]
I updated the ldtk map and that example is now working again. `ldtk_rust` switched to a new json schema in `0.5.3` (main branch is locked at 0.5.2)

`game_of_life` is panicking the same way on the main branch, so that seems unrelated to this PR / bevy 0.7.

**Comment by rparrett** [comment]
Bevy 0.7 was released, so added some commits that hopefully get `bevy_ecs_tilemap` ready for  a release.

If you'd prefer any of that be reverted, squashed, or you'd prefer to handle release details, let me know.

File: README.md [hunk 1]
```diff
@@ -83,5 +83,12 @@ wasm-server-runner .\target\wasm32-unknown-unknown\release\examples\animation.wa
  - Tile flipping by x, y and d, should work for all maps, however "d" (anti diagonal) flipping is not implemented for non-square maps.
  - Besides the above no known issues.
 
+## Bevy Compatibility
+
+|bevy|bevy_ecs_tilemap|
```
Comment (StarArawn) [inline README.md:88]: This is good to have! 👍

**Review by StarArawn — APPROVED** [review]

**Pull request merged**